### PR TITLE
Syntax formatting options

### PR DESCRIPTION
**Pull request by tmat**
Remove fallbacks.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Will do when I get into the office :-)

File: src/Analyzers/CSharp/CodeFixes/UseCollectionExpression/CSharpCollectionExpressionRewriter.cs [hunk 1]
```diff
@@ -50,8 +50,7 @@ internal static class CSharpCollectionExpressionRewriter
 #if CODE_STYLE
         var formattingOptions = CSharpSyntaxFormattingOptions.Default;
 #else
-        var formattingOptions = (CSharpSyntaxFormattingOptions)await workspaceDocument.GetSyntaxFormattingOptionsAsync(
-            fallbackOptions, cancellationToken).ConfigureAwait(false);
+        var formattingOptions = (CSharpSyntaxFormattingOptions)await workspaceDocument.GetSyntaxFormattingOptionsAsync(cancellationToken).ConfigureAwait(false);
```
Comment (CyrusNajmabadi) [inline src/Analyzers/CSharp/CodeFixes/UseCollectionExpression/CSharpCollectionExpressionRewriter.cs:53]: can we remove the #if now?

File: src/Analyzers/CSharp/CodeFixes/UseCollectionExpression/CSharpUseCollectionExpressionForFluentCodeFixProvider.cs [hunk 2]
```diff
@@ -150,8 +150,7 @@ internal partial class CSharpUseCollectionExpressionForFluentCodeFixProvider()
 #if CODE_STYLE
             var formattingOptions = SyntaxFormattingOptions.CommonDefaults;
 #else
-            var formattingOptions = await document.GetSyntaxFormattingOptionsAsync(
-                fallbackOptions, cancellationToken).ConfigureAwait(false);
+            var formattingOptions = await document.GetSyntaxFormattingOptionsAsync(cancellationToken).ConfigureAwait(false);
```
Comment (CyrusNajmabadi) [inline src/Analyzers/CSharp/CodeFixes/UseCollectionExpression/CSharpUseCollectionExpressionForFluentCodeFixProvider.cs:153]: same question

File: src/Features/Core/Portable/CodeFixes/Suppression/AbstractSuppressionCodeFixProvider.PragmaBatchFixHelpers.cs [hunk 3]
```diff
@@ -86,7 +86,7 @@ private static class PragmaBatchFixHelpers
                     properties: diagnostic.Properties,
                     isSuppressed: diagnostic.IsSuppressed);
 
-                var newSuppressionFixes = await suppressionFixProvider.GetFixesAsync(currentDocument, currentDiagnosticSpan, [diagnostic], fallbackOptions, cancellationToken).ConfigureAwait(false);
+                var newSuppressionFixes = await suppressionFixProvider.GetFixesAsync(currentDocument, currentDiagnosticSpan, fallbackOptions, cancellationToken).ConfigureAwait(false);
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/CodeFixes/Suppression/AbstractSuppressionCodeFixProvider.PragmaBatchFixHelpers.cs:89]: error seems legit :)

**Comment by CyrusNajmabadi** [comment]
Overall seems ok.  But will do final pass when this builds :)

File: src/Analyzers/CSharp/CodeFixes/UseCollectionExpression/CSharpUseCollectionExpressionForFluentCodeFixProvider.cs (hunk 2 above)
Comment (tmat) [reply src/Analyzers/CSharp/CodeFixes/UseCollectionExpression/CSharpUseCollectionExpressionForFluentCodeFixProvider.cs:153]: Not Yet.

File: src/Features/Core/Portable/CodeFixes/Suppression/AbstractSuppressionCodeFixProvider.PragmaBatchFixHelpers.cs (hunk 3 above)
Comment (tmat) [reply src/Features/Core/Portable/CodeFixes/Suppression/AbstractSuppressionCodeFixProvider.PragmaBatchFixHelpers.cs]: Yeah, this is where change signature messed up.

File: src/Analyzers/CSharp/CodeFixes/UseCollectionExpression/CSharpCollectionExpressionRewriter.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/Analyzers/CSharp/CodeFixes/UseCollectionExpression/CSharpCollectionExpressionRewriter.cs:53]: ah ok.  good to know :)

File: src/VisualStudio/Core/Def/ExtractClass/VisualStudioExtractClassOptionsService.cs [hunk 4]
```diff
@@ -33,20 +33,17 @@ internal class VisualStudioExtractClassOptionsService : IExtractClassOptionsServ
...
         IThreadingContext threadingContext,
         IGlyphService glyphService,
-        IUIThreadOperationExecutor uiThreadOperationExecutor,
-        IGlobalOptionService globalOptions)
+        IUIThreadOperationExecutor uiThreadOperationExecutor)
     {
         _threadingContext = threadingContext;
         _glyphService = glyphService;
         _uiThreadOperationExecutor = uiThreadOperationExecutor;
-        _globalOptions = globalOptions;
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Def/ExtractClass/VisualStudioExtractClassOptionsService.cs:49]: yes!

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeActions/CodeFixOptionsProvider.cs [hunk 5]
```diff
@@ -68,13 +68,6 @@ private LineFormattingOptions FallbackLineFormattingOptions
         => _fallbackOptions.GetOptions(_languageServices.LanguageServices).CleanupOptions.FormattingOptions.LineFormatting;
 #endif
 
-    private SyntaxFormattingOptions? FallbackSyntaxFormattingOptions
```
Comment (CyrusNajmabadi) [inline src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeActions/CodeFixOptionsProvider.cs:71]: yes!

File: src/Workspaces/VisualBasic/Portable/Formatting/VisualBasicSyntaxFormatting.vb [hunk 6]
```diff
@@ -35,8 +35,8 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Formatting
             End Get
         End Property
 
-        Public Overrides Function GetFormattingOptions(options As IOptionsReader, fallbackOptions As SyntaxFormattingOptions) As SyntaxFormattingOptions
-            Return New VisualBasicSyntaxFormattingOptions(options, DirectCast(fallbackOptions, VisualBasicSyntaxFormattingOptions))
+        Public Overrides Function GetFormattingOptions(options As IOptionsReader) As SyntaxFormattingOptions
+            Return New VisualBasicSyntaxFormattingOptions(options, fallbackOptions:=Nothing)
```
Comment (CyrusNajmabadi) [inline src/Workspaces/VisualBasic/Portable/Formatting/VisualBasicSyntaxFormatting.vb:39]: does this indicate a possible bug?

**Review by CyrusNajmabadi — APPROVED** [review]

**Review by CyrusNajmabadi — APPROVED** [review]

**Pull request merged**